### DISCUSSION TITLE
Email based workflow

### DISCUSSION THREAD

**Discussion by hauleth** [jj-vcs/jj#3350] [Q&A]
I am getting used to [email based workflow](https://git-send-email.io) in Git I wanted to test out `jj`, however from what I see it is not supported (except for the Git compatibility layer). Are there any plans to support such workflow natively?

**Comment by martinvonz** [comment]
I don't personally have any plans to implement it, but I wouldn't reject patches from someone else.

Does it work well for you to use regular `git` commands in a [colocated](https://martinvonz.github.io/jj/v0.15.1/git-compatibility/#co-located-jujutsugit-repos) repo until we have native support?

**Reply by hauleth** [reply]
I haven't tried `jj` jet, so I cannot tell. I am planning to test it out soon.

**Reply by martinvonz** [reply]
I see. You can run regular `git` commands in a colocated repo, so I expect it to work pretty well. It's just that you're pretty much constantly in a detached HEAD state, but hopefully that doesn't impact the email-sending commands (I don't know if they use the current branch to generate a header or something).

**Reply by Zambito1** [reply]
I have used a colocated repo with an email based workflow; it works fine with treating the detatched HEAD as the "working branch" for generating patches.

That said I would also appreciate native patch support if anyone is able to implement it :)

**Reply by BatmanAoD** [reply]
Of note: it appears that `jj diff --git` actually generates output close enough to `git diff --patch` to use with `git apply`.

**Reply by SamuelV7** [reply]
Has this been implemented yet? If not, I'm happy to work on it and submit a patch

**Reply by PhilipMetzger** [reply]
Work for it is tracked in #4438 although that largely depends on #2702
